test/gtest/fusion_test.cpp

```diff
@@ -0,0 +1,153 @@
+/*******************************************************************************
+ *
+ * MIT License
+ *
+ * Copyright (c) 2023 Advanced Micro Devices, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+#include <gtest/gtest.h>
+#include <miopen/miopen.h>
+
+#if MIOPEN_BACKEND_HIP // equiv to hipInit(0)?
+
+std::vector<int> strides(std::vector<int> dims)
+{
+    std::vector<int> strides_array(4);
+    int d = 1;
+    for(int i = 3; i >= 0; --i)
+    {
+        strides_array[i] = d;
+        d *= dims[i];
+    }
+    return strides_array;
+}
+
+void upload(float*& pd, const std::vector<float>& v)
+{
+    hipMalloc(&pd, v.size() * 4);
+    hipMemcpy(pd, &v[0], v.size() * 4, hipMemcpyHostToDevice);
+}
```

Comment:
You can use the `tensor` class in the tests header. Also dont use the hip calls directly, instead use `miopen::Handle` methods.

Reply:
done